crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/cheated_syscalls.rs

```diff
@@ -47,15 +48,18 @@ pub fn get_execution_info_syscall(
     _remaining_gas: &mut u64,
 ) -> SyscallResult<GetExecutionInfoResponse> {
     let execution_info_ptr = syscall_handler.get_or_allocate_execution_info_segment(vm)?;
+    let current_call_stack = &mut runtime_state.cheatnet_state.trace_data.current_call_stack;
 
-    let contract_address = syscall_handler.storage_address();
+    let cheated_data = if current_call_stack.size() == 1 {
+        CheatedData::new(
+            runtime_state.cheatnet_state,
+            &syscall_handler.storage_address(),
+        )
```

Comment:
This way every `get_caller_address` etc. done directly in the test code will count as one call for the purpose of our cheatcodes (the constructor modifies the cheatnet state, commented below), I don't think we want this. In fact there is no way to call the test code contract from any other contract: 

https://github.com/foundry-rs/starknet-foundry/blob/9a1b4bc0c85c99f2ec21eb62c0ec99b8ea310e54/crates/cheatnet/src/constants.rs#L63)

I think we should prevent cheating `test_address` with any span but `Indefinite`. Maybe we can do it on Cairo level and make assertions in our Rust code - I don't really like coupling Cairo and Rust that much, but I have no better idea, feel free to explore possible solutions.
cc @Arcticae

Reply:
Why is this not what we want? 

Maybe it's not directly visible without tests, but the current behaviour of `CheatTarget::All`, is cheating each address n times separately. It does not share `CheatSpan`'s between addresses.

// This has been refactored via the rest of the comments, but the logic stayed the same

Reply:
> Maybe it's not directly visible without tests, but the current behaviour of CheatTarget::All, is cheating each address n times separately. It does not share CheatSpan's between addresses.

I don't understand what you mean, don't we set a separate var with global cheat? Also I don't see how is it relevant tbh

Nonetheless I still don't think we should decrease the counter if it will be `Indefinite` every time, it is just an operation that does nothing

Reply:
I'm not a huge fan of constraining the domain in specific cases, I'd rather allow for any CheatSpan on test_address, just don't update the cheats

Reply:
[Do not update cheats for test address](https://github.com/foundry-rs/starknet-foundry/pull/1703/commits/1105f407b309c86ee8182aceae4819a84d9fe0d1)

Reply:
>I'm not a huge fan of constraining the domain in specific cases, I'd rather allow for any CheatSpan on test_address, just don't update the cheats

Agree, that's a better idea, we can still constraint them on the Cairo level later on, right?